pastas/stressmodels.py

```diff
@@ -1151,3 +1153,169 @@ def to_dict(self, series=True):
+        self.dmax = dmax
+        super().__init__(prec, evap, rfunc=rfunc, **kwargs)
+
+    def set_init_parameters(self):
+        # parameters for the first drainage level
+        p0 = self.rfunc.get_init_parameters(self.name)
+        one = One(meanstress = self.dmin + 0.5 * (self.dmax - self.dmin))
+        pd0 = one.get_init_parameters(self.name).squeeze()
+        p0.loc['{}_d'.format(self.name)] = pd0
+        p0.index = ['{}0'.format(x) for x in p0.index]
+
+        # parameters for the second drainage level
+        p1 = self.rfunc.get_init_parameters(self.name)
+        initial = self.dmin + 0.75 * (self.dmax - self.dmin)
+        pd1 = Series({'initial':initial, 'pmin':self.dmin, 'pmax':self.dmax,
+                      'vary':True, 'name':self.name})
+        p1.loc['{}_d'.format(self.name)] = pd1
+        p1.index = ['{}1'.format(x) for x in p1.index]
+
+        # parameters for the recharge-method
+        pr = self.recharge.get_init_parameters(self.name)
+
+        # combine all parameters
+        self.parameters = concat([p0, p1, pr])
+
+    def simulate(self, p=None, tmin=None, tmax=None, freq=None, dt=1):
+        stress = self.get_stress(p=p, tmin=tmin, tmax=tmax, freq=freq)
+        h = self.tarso(p[:-self.recharge.nparam], stress.values, dt)
+        sim = Series(h, name=self.name, index=stress.index)
+        return sim
+
+    def to_dict(self, series=True):
+        data = super().to_dict(series)
+        data['dmin'] = self.dmin
+        data['dmax'] = self.dmax
+        return data
+
+    @staticmethod
+    def check_stressmodel_compatibility(ml):
+        """Internal method to check if no other stressmodels, a constants or a
+        transform is used."""
+        if len(ml.stressmodels)>1:
+            msg = ('A TarsoModel cannot be combined with other stressmodels. '
+                   'Either remove the TarsoModel or the other stressmodels.')
+            logger.warning(msg)
+        if ml.constant is not None:
+            msg = ('A TarsoModel cannot be combined with a constant. '
+                   'Either remove the TarsoModel or the constant.')
+            logger.warning(msg)
+        if ml.transform is not None:
+            msg = ('A TarsoModel cannot be combined with a transform. '
+                   'Either remove the TarsoModel or the transform.')
+            logger.warning(msg)
+
+    @staticmethod
+    @njit
+    def tarso(p, r, dt):
+        """Calculates the head based on exponential decay of the previous
+        timestep and recharge, using two thresholds."""
+        A0, a0, d0, A1, a1, d1 = p
+
+        # calculate physical meaning of these parameters
+        S0 = a0 / A0
+        c0 = A0
+
+        S1 = a1 / A1
+        c1 = A1
+
+        # calculate effective parameters for the top level
+        c_e = 1 / ((1/c0) + (1/c1))
+        d_e = (c1 / (c0+c1)) * d0 + (c0 / (c0+c1)) *d1
+        a_e = S1 * c_e
+
+        h = np.full(len(r), np.NaN)
+        for i in range(len(r)):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/pastas/pastas/pullRequest?prid=5245245)